teps/0062-catalog-tags-and-hub-categories-management.md

```diff
@@ -0,0 +1,232 @@
+---
+status: proposed
+title: Catalog Tags and Hub Categories Management
+creation-date: '2021-03-30'
+last-updated: '2021-03-30'
+authors:
+  - '@piyush-garg'
+  - '@PuneetPunamiya'
+---
+
+# TEP-0062: Catalog Tags and Hub Categories Management
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+
+- [Proposed Solution](#proposed-solution)
+  - [Searching v/s Filtering](#searching-v/s-filtering)
+  - [Example of Operator Hub](#example-of-operator-hub)
+  - [Pros of Having Categories](#pros-of-having-categories)
+  - [Reason Behind Not Using Tags Only](#reason-behind-not-using-tags-only)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+The Tekton catalog is a collection of blessed Tekton resources that can be used with any system that supports the Tekton API. Each Tekton resource has tags as a field in annotation. All tags in catalog resources are mapped to categories in Hub. Currently when new tags are found in catalog, these tags are mapped to categories called _others_. Hub needs to check every time and do the maintenance of mapping the tag to categories so that it does not fall under Others.
+
+This is [config.yaml](https://github.com/tektoncd/hub/blob/main/config.yaml) in Hub where we map tags to the categories
+
+```yaml
+---
+categories:
+  - name: Build Tools
+    tags: [build-tool]
+  - name: CLI
+    tags: [cli]
+  - name: Cloud
+    tags: [gcp, aws, azure, cloud]
+  - name: Deploy
+    tags: [deploy]
+  - name: Others        👈 Any new tags are mapped here
+    tags: []
+  - name: Test Framework
+    tags: [test]
+```
+
+## Motivation
+
+### Goals
+
+- Make the manual work automated and also properly do the mapping of the resources.
+
+### Non-Goals
+
+### Use Cases (optional)
+
+- Filter the resources on Hub using categories
+- Search the resources based on tags as keywords
+
+## Requirements
+
+- Add category field as annotation in a task
+
+For e.g.
```

Comment:
NIT: replace "for e.g." with "E.g." or "For example"
- e.g. means "for example" (exempli gratia)